dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/constraints/DatadogDataConstraints.kt

```diff
@@ -97,24 +116,17 @@ internal class DatadogLogConstraints : LogConstraints {
 
     // region Internal/Attribute
 
-    private val attributeKeyTransforms = listOf<StringTransform>(
-        // A key can only have 10 levels
-        { key ->
-            var dotCount = 0
-            val mapped = key.map {
-                if (it == '.') {
-                    dotCount++
-                    if (dotCount > 9) '_' else it
-                } else it
-            }
-            String(mapped.toCharArray())
-        }
-    )
-
-    private fun convertAttributeKey(rawKey: String?): String? {
-        return attributeKeyTransforms.fold(rawKey) { key, transform ->
-            if (key == null) null else transform.invoke(key)
+    private fun convertAttributeKey(rawKey: String, keyPrefix: String?): String {
+        // prefix = "a.b" => dotCount = 1+1 ("a.b." + rawKey)
+        var dotCount =
+            keyPrefix?.let { it.count { character -> character == '.' } + 1 } ?: 0
```

Comment:
we can pass `dotCount` with a meaningful name to this function as it doesn't change in `validateAttributes` function.
we can compute it once in `validateAttributes` and pass it to `convertAttributeKey`.

if you think this method won't cause performance issues, we can keep it as is; because the current code is very clear to me as a reader 👍

Reply:
oh good point. Will add this improvement ;)